src/components/TresCanvas.vue

```diff
@@ -104,14 +108,41 @@ const canvas = ref<HTMLCanvasElement>()
 */
 const scene = shallowRef<TresScene | Scene>(new Scene())
 
-const instance = getCurrentInstance()?.appContext.app
+const instance = getCurrentInstance()
 extend(THREE)
 
 const createInternalComponent = (context: TresContext, empty = false) =>
   defineComponent({
     setup() {
       const ctx = getCurrentInstance()?.appContext
-      if (ctx) { ctx.app = instance as App }
+      if (ctx) { ctx.app = instance?.appContext.app as App }
+      const provides = {}
+
+      // Helper function to recursively merge provides from parents
+      function mergeProvides(currentInstance: any) {
+        if (!currentInstance) { return }
+
+        // Extract provides from the current instance and merge them
+        if (currentInstance.provides) {
+          Object.assign(provides, currentInstance.provides)
```

Comment:
Precedence is backwards here.

If the same key is provided more than once in the ancestor chain, the most distant ancestor is used. But it should be the closest.

Here's a set of files that shows the error.

* Save each code block to the filename preceding it.
* Open `index.vue`
* Open the console.

index.vue

```vue
<script setup lang="ts">
import { provide } from 'vue'
import TheExperience from './TheExperience.vue'

provide('test', '❌ Precendence is incorrect')
</script>

<template>
  <TheExperience />
</template>
```

TheExperience.vue

```vue
<script setup lang="ts">
import { TresCanvas } from '@tresjs/core'
import { provide } from 'vue'
import ComponentWithInject from './ComponentWithInject.vue'

provide('test', '✅ Precedence is correct')
</script>

<template>
  <TresCanvas>
    <ComponentWithInject />
  </TresCanvas>
</template>
```

ComponentWithInject.vue

```vue
<script setup lang="ts">
import { inject } from 'vue'

console.log(inject('test'))
</script>

<template>
  <TresMesh>
    <TresBoxGeometry />
    <TresMeshNormalMaterial />
  </TresMesh>
</template>
```